src/parsers/doctype.js

```diff
@@ -0,0 +1,45 @@
+const ENTITY_RE = /<!ENTITY\s+([\w.]*)\s+("[^"]*"|'[^']*')\s*>/;
+
+
+export default class DoctypeParser {
+  /*
+   * Minimalistic parser for DTD files.
+   * We are using DTD files in our language packs to specify translations.
+   *
+   * It doesn't do any proper XML parsing because our DTD files are only
+   * using a very small subset of what's needed so a simple regular expression
+   * works just fine.
+   *
+   * The parsing code is largely inspired by scattered code from mozilla-central.
+   */
+
+  constructor(dtdString, collector, { filename = null } = {}) {
+    this._dtdString = dtdString;
+    this.collector = collector;
+    this.filename = filename;
+    this.isValid = null;
+  }
+
+  parse() {
+    const entities = this._dtdString.match(new RegExp(ENTITY_RE, 'g'));
```

Comment:
Could it be `<!entity` as well? If so, we might want to add the `i` flag.

Reply:
hmm, good point. I have not found anything that is lowercase and I'm not even sure about the spec here but I think it's better to be safe than sorry :)

I'll add that.